### ISSUE TITLE
calling runtime.ReadMemStats freezes runtime while any goroutines are in a non pre-emptable tight loop

### ISSUE THREAD

**Issue by foubarre** [golang/go#28289]
Please answer these questions before submitting your issue. Thanks!

### What version of Go are you using (`go version`)?
go1.11.1 linux/amd64

### Does this issue reproduce with the latest release?
yes

### What operating system and processor architecture are you using (`go env`)?
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/xxx/.cache/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/xxx/go"
GOPROXY=""
GORACE=""
GOROOT="/home/xxx/go1.11.1"
GOTMPDIR=""
GOTOOLDIR="/home/xxx/go1.11.1/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/home/xxx/tmp/go-build973912040=/tmp/go-build -gno-record-gcc-switches"

### What did you do?
```
package main

import (
	"fmt"
	"runtime"
	"sync"
	"time"
)

func main() {
	defer fmt.Println("Bye, playground")
	fmt.Println("Hello, playground")
	go func() {
		for {
			fmt.Println(".")
			time.Sleep(100 * time.Millisecond)
		}
	}()
	var wg sync.WaitGroup
	wg.Add(1)
	go func() {
		first := true
		for {
			if first {
				first = false
				wg.Done()
			}
			// Uncommenting sleep call gives expected behavior
			// time.Sleep(time.Nanosecond)
		}
	}()
	wg.Wait()
	fmt.Println("Hello, ReadMemStats")
	var m runtime.MemStats
	runtime.ReadMemStats(&m)
	fmt.Println("Bye, ReadMemStats")
	time.Sleep(time.Second)
}
```

### What did you expect to see?
```
ReadMemStats call completes successfully:Hello, playground
.
Hello, ReadMemStats
Bye, ReadMemStats
.
.
.
.
.
.
.
.
.
Bye, playground
```

### What did you see instead?
all goroutines seem frozen after ReadMemStats is called:
```
Hello, playground
.
Hello, ReadMemStats
```

**Comment by foubarre** [comment]
The empty for loop example is contrived but we do have fairly long running data crunching tight loops  in a prod system which is also calling ReadMemStats on a regular basis for logging purposes.
I haven't seen this specific issue mentioned before so creating it for tracking purposes and getting confirmation #24543 is expected to fix it.

**Comment by ianlancetaylor** [comment]
This is true and essentially a restatement of #10958.  Closing as dup.  (There is an active attempt to fix this problem in #24543.)
